test/Microsoft.Extensions.DataIngestion.Tests/DocumentTokenChunker/DocumentTokenChunker.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.ML.Tokenizers;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.DataIngestion.Tests
+{
+    internal class DocumentTokenChunker : DocumentChunker
+    {
+        private readonly Tokenizer _tokenizer;
+        private readonly int _chunkSize;
+        private readonly int _chunkOverlap;
+        public DocumentTokenChunker(Tokenizer tokenizer, int chunkSize, int chunkOverlap)
+        {
+            if (chunkOverlap >= chunkSize)
+                throw new ArgumentException("Chunk overlap must be less than chunk size.", nameof(chunkOverlap));
+
+            _tokenizer = tokenizer ?? throw new ArgumentNullException(nameof(tokenizer));
+            _chunkSize = chunkSize > 0 ? chunkSize : throw new ArgumentOutOfRangeException(nameof(chunkSize));
+            _chunkOverlap = chunkOverlap >= 0 ? chunkOverlap : throw new ArgumentOutOfRangeException(nameof(chunkOverlap));
+        }
+
+        public override ValueTask<List<Chunk>> ProcessAsync(Document document, CancellationToken cancellationToken = default)
+        {
+            if (document is null) throw new ArgumentNullException(nameof(document));
+
+            var tokens = _tokenizer.EncodeToIds(document.Markdown);
+            var token_groups = CreateGroups(tokens);
```

Comment:
This code will one day be moved to a repo where using `var` when the type is not obvious is prohibited. So let's not use `var` is such situations, as it would require us to perform some extra work when moving the code in the future.

https://github.com/dotnet/extensions/blob/main/CONTRIBUTING.md#dos-and-donts

leads to:

https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md